netbox/netbox/middleware.py

```diff
@@ -166,3 +168,43 @@ def _get_groups(self, request):
             groups = []
         logger.debug(f"Groups are {groups}")
         return groups
+
+
+class MaintenanceModeMiddleware:
+    """
+    Middleware that checks if the application is in maintenance mode
+    and restricts write-related operations to the database.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if getattr(settings, 'MAINTENANCE_MODE', False):
```

Comment:
We need to reference this setting from `get_config()` because it's a dynamic parameter, which can be set either in `configuration.py` or a config revision in the database.

```suggestion
        if get_config().MAINTENANCE_MODE:
```